validator/src/main/java/com/amazon/aoc/validators/ContainerInsightStructuredLogValidator.java

```diff
@@ -34,18 +34,39 @@
           "PodNet"
   );
 
+  //expected template Array for containerd runtime
+  private static final List<String> LOG_TYPE_CONTAINERD = Arrays.asList(
+          "Cluster",
+          "ClusterNamespace",
+          "ClusterService",
+          "Container",
+          "Node",
+          "NodeDiskIO",
+          "NodeFS",
+          "NodeNet",
+          "Pod",
+          "PodNet"
+  );
+
   private static final int MAX_RETRY_COUNT = 15;
   private static final int QUERY_LIMIT = 100;
+  private static List<String> LOG_TYPE_TO_VALIDATE;
 
   @Override
   void init(Context context, FileConfig expectedDataTemplate) throws Exception {
     logGroupName = String.format("/aws/containerinsights/%s/performance",
             context.getCloudWatchContext().getClusterName());
     MustacheHelper mustacheHelper = new MustacheHelper();
+    if (context.getTestcase().contains("containerinsight_eks")) {
```

Comment:
good catch.
I think we can refactor this a bit so that we don't need to have this branching behavior.

* Lets create a base class `ContainerInsightStructuredLogValidatorBase` that contains everything currently here, except the LOG_TYPE_TO_VALIDATE. This class should have an abstract method `getLogTypeToValidate`, that will return the `List<String>` containing that information.
* Create a class ContainerInsightStructuredLogValidatorDocker that implements:
```
@override
public List<String> getLogTypeToValidate() {
return Arrays.asList(
          "Cluster",
          "ClusterNamespace",
          "ClusterService",
          "Container",
          "ContainerFS",
          "Node",
          "NodeDiskIO",
          "NodeFS",
          "NodeNet",
          "Pod",
          "PodNet"
  );
}
```
* Create a class ContainerInsightStructuredLogValidatorContainerd that implements:
```
@override
public List<String> getLogTypeToValidate() {
return Arrays.asList(
          "Cluster",
          "ClusterNamespace",
          "ClusterService",
          "Container",
          "Node",
          "NodeDiskIO",
          "NodeFS",
          "NodeNet",
          "Pod",
          "PodNet"
  );
}
```
* Update the ValidatorFactory accordingly creating distinct validation types (`containerinsight-eks-docker-logs` and `containerinsight-eks-containerd-logs`.
* Update testcases accordingly to use the new validator.

Reply:
I have resolved it with modified case name.

Reply:
I've sat on this idea a bit more. I don't think the pattern of using the test case name to make decisions is a good idea. I know there are examples elsewhere, in terraform and possibly the validator, but I don't think we should add more functionality that depends on it. It is not apparent when troubleshooting and when not documented properly it can cause a lot of headache. I would prefer that a specific variable gets passed in when necessary to the validator. One that is explicitly named so that we know that we are making a decision based on it.